File: Scripts/GunShot.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

[RequireComponent(typeof(AudioSource))]
public class GunShot : MonoBehaviour {
	void Update () {
		if (Input.GetButtonDown ("Fire1")) {
		    AudioSource gunsound = GetComponent<AudioSource>();
			gunsound.Play();
			GetComponent<Animation>().Play("GunFire");
			GlobalAmmo.CurrentAmmo -= 1;
		}	
	}
}

File: Scripts/GunShot.js
var GunShotSound : AudioSource;
var flash : GameObject;

function Update () { 
	
	if(GlobalAmmo.LoadedAmmo >= 1){
		//Edit/ProjectSettings/Input
		if(Input.GetButtonDown("Fire1")) {
			
			GunShotSound.Play();
			flash.SetActive(true);
			MuzzleOff();
			GetComponent.<Animation>().Play("GunShot");
			GlobalAmmo.LoadedAmmo -= 1;
		}
	}

}
function MuzzleOff(){
	yield WaitForSeconds(0.1);
	flash.SetActive(false);
}
